packages/fuels-core/src/code_gen/abigen.rs

```diff
@@ -124,21 +124,22 @@ impl Abigen {
                     use fuels::core::{EnumSelector, Parameterize, Tokenizable, Token, try_from_bytes};
                     use fuels::signers::LocalWallet;
                     use fuels::tx::{ContractId, Address};
+                    use fuels::types::bech32::Bech32ContractId;
                     use fuels::types::errors::Error as SDKError;
                     use fuels::types::param_types::{EnumVariants, ParamType};
                     use std::str::FromStr;
                 },
                 quote! {
                     pub struct #name {
-                        contract_id: ContractId,
-                        wallet: LocalWallet
+                        pub contract_id: Bech32ContractId,
+                        pub wallet: LocalWallet
```

Comment:
Did we agree to make the `contract_id` and `wallet` public ? If so we do not need the getters anymore.

Reply:
Otherwise LGTM! I like how it was implemented!

Reply:
This was done before your recent PR. I will update it to use the getters. However, I think we need to open up this discussion at some point again. We added an instance builder to abigen to avoid collision with `new` but now we have the same problem with the getters/setter and soon with `instance::connect` too and we still kept the builder.

Reply:
I agree.